mozdownload/scraper.py

```diff
@@ -611,19 +618,38 @@ def build_filename(self, binary):
                            'STUB': '-stub' if self.is_stub_installer else '',
                            'EXT': self.extension}
 
+    def get_build_info(self):
+        self.version = self.query_versions(self.version)[0]
+
+    def query_versions(self, version=None):
+        """Check specified version and resolve special values"""
+        if version not in RELEASE_AND_CANDIDATE_LATEST_VERSIONS:
+            return [version]
+
+        url = urljoin(self.base_url, 'releases/')
+        parser = self._create_directory_parser(url)
+        if version:
+            versions = parser.filter(RELEASE_AND_CANDIDATE_LATEST_VERSIONS[version])
+            from distutils.version import LooseVersion
+            versions.sort(key=LooseVersion)
+            return [versions[-1]]
+        else:
+            return parser.entries
+
 
 class ReleaseCandidateScraper(ReleaseScraper):
     """Class to download a release candidate build from the Mozilla server"""
 
-    def __init__(self, version, build_number=None, *args, **kwargs):
-        self.version = version
+    def __init__(self, build_number=None, *args, **kwargs):
         self.build_number = build_number
 
```

Comment:
Please be aware that the Candidate builds are special release builds. That's why the `ReleaseCandidateScraper` class is based on the `ReleaseScraper` class. Don't ask me why we didn't call its base `__init__()` method, but the one from the `Scraper` class. So that is something I have changed here. Means the version is still mandatory and as you noticed will be picked up by the base class which is `ReleaseScraper`. Regarding other scrapers, none of those need a version but a branch. As result the factory class exists to make it easier to work with mozdownload. It will only pick those parameters which are necessary and known by a class. The others will not be obeyed.